### PR TITLE
Backyourstack: Fix broken test

### PR DESCRIPTION
**Pull request by flickz**
Resolve https://github.com/opencollective/opencollective/issues/2648

### CODE REVIEW

**Comment by znarf** [comment]
The issue was asking to also cover a failure. Make sure it's properly handled. Can you check that?

**Comment by flickz** [comment]
I'm still on it.

**Comment by znarf** [comment]
I don't get the last change, I expected to see a new test covering that a failure is properly handled.

**Comment by flickz** [comment]
I thought the goal was to be able to detect error that occurred during the process of dispatching, instead of having errors ignored while the tests continue to pass. 

This change will cause the test to fail whenever an error occurs.

**Comment by flickz** [comment]
> I don't get the last change, I expected to see a new test covering that a failure is properly handled.

I understand now, I'll add additional test

File: test/graphql.createOrder.backyourstack.test.js [hunk 1]
```diff
@@ -187,10 +194,21 @@ describe('createOrder', () => {
...
+  });
+
+  it('returns  error when it is not time for dispatch', async () => {
     // When the order is created
     const res = await utils.graphqlQuery(backyourstackDispatchOrder, { id: orderCreated.id }, xdamman);
-    // There should be no errors
-    res.errors && console.error(res.errors);
-    expect(res.errors).to.not.exist;
+    // There should be an error
+    expect(res.errors).to.exist;
```
Comment (znarf) [inline test/graphql.createOrder.backyourstack.test.js]: Can you test the error message?

File: server/lib/backyourstack/dispatcher.js [hunk 2]
```diff
@@ -176,6 +176,9 @@ export async function dispatch(order, subscription) {
       },
     });
   } catch (err) {
+    if (process.env.NODE_ENV !== 'production') {
+      throw err;
+    }
```
Comment (znarf) [inline server/lib/backyourstack/dispatcher.js]: We should not have a specific behavior like that for production. Remove it or make it the default.

**Pull request merged**